sawmill-core/src/main/java/io/logz/sawmill/http/ExternalMappingsClient.java

```diff
@@ -64,6 +60,40 @@ public ExternalMappingsClient(ExternalMappingSourceProcessor.Configuration confi
         return mappings;
     }
 
+    private void loadMappingsFromInputStream(InputStream inputStream, Map<String, Iterable<String>> mappings) throws IOException {
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(inputStream))) {
+            int linesCount = 0;
+            String inputLine;
+            while ((inputLine = in.readLine()) != null) {
+                if (inputLine.isEmpty()) continue;
+
+                if (++linesCount > ExternalMappingSourceProcessor.Constants.EXTERNAL_MAPPING_MAX_LINES) {
+                    throw new ProcessorInitializationException(
+                        String.format("Cannot load external mappings, the mapping length exceeds the limit of %d lines",
+                            ExternalMappingSourceProcessor.Constants.EXTERNAL_MAPPING_MAX_LINES)
+                    );
+                }
+
+                inputLine = StringEscapeUtils.escapeJava(inputLine);
+                Pair<String, Iterable<String>> entry = toKeyValuePair(inputLine);
+                mappings.merge(entry.getLeft(), entry.getRight(), Iterables::concat);
+            }
+        }
+    }
+
+    private void ensureContentLengthIsBelowTheLimit(HttpURLConnection conn) {
+        String contentLength = conn.getHeaderField(HttpHeaders.CONTENT_LENGTH);
+        if(contentLength == null) return;
```

Comment:
can this be null while the file is still valid and big?

Reply:
yes, the "Content-Length" header is not required by HTTP protocol. 
I cannot assume it's always present.

Reply:
So if the url is not configured to return this header, this validation is skipped.
So if you have a file with "only" 50,000 lines, but each line is more than 1MB, we will pass EXTERNAL_MAPPING_MAX_BYTES

Reply:
yep, but there is nothing we can do to avoid this.

for s3 objects Content-Length is always present
for other cases we can instruct users to add `Content-Lenght` header to protect their systems

Reply:
there are two things we can do:
1. do not allow urls without that header (makes things more complex for users, but protects our system better)
2. add some validation while iterating the file - like we count the lines, we can roughly count the bytes/chars per line

Reply:
added fallback for counting length of the input if Content-Lenght is absent

Reply:
I wouldn't do it as a fallback, I would just remove the header-based validation.
Remember that anyone can manually tweak that header and send you fake length value.
Also, this will keep things simpler.

Reply:
uh, this will require adding ResponseTransformer to the tests. Hold on.

Reply:
you anyway needed to test the fallback, no?

Reply:
try finding a way to override the constants in the test